### PR TITLE
Setup: Cleanup Cobbler setup

### PR DESCRIPTION
**Pull request by SchoolGuy**
Fixes https://github.com/SUSE/spacewalk/issues/15381

## What does this PR change?

- Remove Cobbler 2.X.X setup scripts (not used anymore)
- Fix a [bug](https://github.com/SUSE/spacewalk/issues/15381) which leads to a case where EFI servers are not installable

> I will update this once I am done with all changes to the appropriate scripts.

## GUI diff

No difference.

## Documentation
- No documentation needed: **add explanation. This can't be used if there is a GUI diff**
- No documentation needed: only internal and user invisible changes
- Documentation issue was created: [Link for SUSE Manager contributors](https://github.com/SUSE/spacewalk/issues/new?template=ISSUE_TEMPLATE_DOCUMENTATION.md&labels=documentation&projects=SUSE/spacewalk/31), [Link for community contributors](https://github.com/uyuni-project/uyuni-docs/issues/new).
- (OPTIONAL) [Documentation PR](https://github.com/uyuni-project/uyuni-docs/pulls)

- [ ] **DONE**

## Test coverage
- No tests: **add explanation**
- No tests: already covered
- Unit tests were added
- Cucumber tests were added

- [ ] **DONE**

## Links

Fixes #
Tracks # **add downstream PR, if any**

- [ ] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by mcalmer** [comment]
JFYI: cobbler 20 was only used by original spacewalk. In older SUMA we used already 2.2 and 2.6 which was never configured via this special script. For that we always had spacewalk-cobbler-setup.

**Comment by SchoolGuy** [comment]
@mcalmer If possible I would love to get your input on my changes.

**Review by meaksh — COMMENTED** [review]
It seems you are missing a changelog entry for `spacewalk-setup` package.

Another question, as you mentioned in Rocket.Chat, it seems the old script has `--help` and somehow it seems you can manually call it from the CLI.

Are we currently losing such functionality?

I'm not expert on this area so not really sure which are the real use cases we have for this script.

**Comment by mcalmer** [comment]
@SchoolGuy Check `Makefile.PL` - it contains references to cobbler20-setup and spacewalk-setup-cobbler.
As it is a special Makefile for perl, I do not exactly know if it will try to generate man pages from it and will break when the file is now in python. I would suggest you try to build that package manually and see how that behave.

**Comment by mcalmer** [comment]
https://github.com/uyuni-project/uyuni/pull/4030/files#diff-0ccbf54ec6c63979ce854bc1f3bfff6928637d55049132b6f3e661280ef9b72aR186

There is a pod2man call on spacewalk-setup-cobbler in the spec. That will not work as well. We need to replace the man page somehow.

**Comment by SchoolGuy** [comment]
Okay so things to do:

- [ ] Replace man page
- [ ] Readd help functionality
- [ ] Add changelog entry for spacewalk-setup

**Review by agraul — COMMENTED** [review]
I've only skimmed the new code so far and a few minor things caught my eye. Comments are inline. I'm a bit picky, but nitpicks are marked as such, ignore them if you want.

File: spacewalk/setup/bin/spacewalk-setup-cobbler [hunk 1]
```diff
@@ -1,307 +1,113 @@
-#!/usr/bin/perl
+#!/usr/bin/python
```
[the remark this answers is not in the document]
Comment (SchoolGuy) [reply spacewalk/setup/bin/spacewalk-setup-cobbler]: Not at all. Adjusted it.

File: spacewalk/setup/bin/spacewalk-setup-cobbler [hunk 2]
```diff
@@ -1,307 +1,113 @@
...
+
+def backup_file(file_path: str):
+    """
+    Copies the file and adds a suffix ".backup" to it.
+    
+    :param file_path: The full path to the file which should be backed up.
+    :raises FileNotFoundError: In case the path specified was not existing.
+    """
+    if not os.path.exists(file_path):
+        raise FileNotFoundError("file at path \"%s\" not found" % file_path)
```
[the remark this answers is not in the document]
Comment (SchoolGuy) [reply spacewalk/setup/bin/spacewalk-setup-cobbler]: Okay if this is not needed I will remove it.

File: spacewalk/setup/bin/spacewalk-setup-cobbler [hunk 3]
```diff
@@ -1,307 +1,113 @@
...
+    """
+    Manipulate the main Cobbler configuration file which is in YAML format. This function backs the original
+    configuration up and writes a new one with the required changes to the disk.
+    
+    :param config_dir: The directory of Cobbler where the config files are.
+    :param settings_yaml: The name of the main YAML file of Cobbler.
+    """
+    full_path = os.path.join(config_dir, settings_yaml)
+    backup_file(full_path)
+    # Read YAML
```
[the remark this answers is not in the document]
Comment (SchoolGuy) [reply spacewalk/setup/bin/spacewalk-setup-cobbler]: Removed. Sounds reasonable.

**Review by vzhestkov — APPROVED** [review]
Looks good

File: spacewalk/setup/bin/spacewalk-setup-cobbler [hunk 4]
```diff
@@ -1,307 +1,121 @@
...
+    # Modify Authentication to Spacewalk module
+    cp["authentication"]["module"] = "authentication.spacewalk"
+    # Write Ini File
+    with open(full_path, "w") as modules_conf:
+        cp.write(modules_conf)
+
+
+def remove_virtual_host(conf_directory: str, conf_file: str):
+    """
+    Replaces all occurences of lines with "VirtalHost" with an empty one.
```
Comment (aaannz) [inline spacewalk/setup/bin/spacewalk-setup-cobbler:86]:
```suggestion
    Replaces all occurrences of lines with "VirtualHost" with an empty one.
```

**Review by agraul — APPROVED** [review]
LGTM!

**Comment by sbluhm** [comment]
@SchoolGuy , unfortunately, this PR causes an issue on Enterprise Linux. `spacewalk-setup-cobbler` uses 
` --apache2-config-directory', '-a', dest='httpd_config_directory', default="`**/etc/httpd**`"`
Correct location on EL is `/etc/httpd/conf.d`
What exactly is the purpose of this line if it always gets overwritten anyways?
https://github.com/uyuni-project/uyuni/blob/32c6a635be497454fe6b7ad4aea46c02f52d2dfc/spacewalk/setup/bin/spacewalk-setup-cobbler#L30
(and in similar context L28 for `COBBLER_CONFIG_DIRECTORY`)

The values will always get overwritten by
https://github.com/uyuni-project/uyuni/blob/32c6a635be497454fe6b7ad4aea46c02f52d2dfc/spacewalk/setup/bin/spacewalk-setup-cobbler#L101

**Pull request merged**